### PR TITLE
ref(codeowners) Update CODEOWNERS for visibility team

### PR DESCRIPTION
**Pull request by evanh**
Add performance/discover ownership rules for code owners.

### CODE REVIEW

**Review by dashed — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -77,3 +77,40 @@ Makefile                    @getsentry/owners-sentry-dev
...
+
+### Perfomance/Discover ###
+/src/sentry/snuba/                                              @getsentry/visibility
+/tests/sentry/snuba/test_discover.py                            @getsentry/visibility
+
+/src/sentry/discover/                                           @getsentry/visibility
+
+/src/sentry/utils/snuba.py                                      @getsentry/visibility
+
+/src/sentry/api/event_search.py                                 @getsentry/visibility
```
Comment (wmak) [inline .github/CODEOWNERS:90]: Hmm `api/event_search.py` might be bigger than just visibility and [the last matching pattern takes the most precedence.](https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners)

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -77,3 +77,40 @@ Makefile                    @getsentry/owners-sentry-dev
...
+/src/sentry/api/issue_search.py                                 @getsentry/visibility
```
Comment (markstory) [inline .github/CODEOWNERS:91]: I'm ok with taking on issue search, but it is a bigger scope item.

Comment (evanh) [reply .github/CODEOWNERS:91]: Oh I didn't realize the precedence rules. I assumed multiple people could own a file. I'll remove issue_search.

File: .github/CODEOWNERS (hunk 1 above)
Comment (evanh) [reply .github/CODEOWNERS:90]: Well we can fight with whoever else wants it I guess. If it's an issue we can make a new team called "search" or something and use that.

File: .github/CODEOWNERS (hunk 2 above)
Comment (evanh) [reply .github/CODEOWNERS:91]: Actually, maybe we should keep it. I don't know if there's a team that actually owns it right now.

Comment (markstory) [reply .github/CODEOWNERS:91]: I don't think there is an alternate owner for issue search right now. We might as well take it as interim coverage.

**Comment by joshuarli** [comment]
Nice, thank you to those involved for adding additional coverage!

**Pull request merged**